Bottom: 5553795186fb1aefed2fa1bac1a4aff911b5d85c
Top:    1a578b445ba8934454cf4a7561bfa3f2c1aef7b8
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-10-04 18:19:01 -0700

Refresh of allow-arbitrary-python-streams

---
diff --git a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
index 2709adf37e6..9c716617201 100644
--- a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
+++ b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
@@ -132,14 +132,6 @@ enum class PyInitialValue { Invalid, Empty };
 
 class PythonObject {
 public:
-  static Error nullDeref() {
-    return llvm::createStringError(llvm::inconvertibleErrorCode(),
-                                   "A NULL PyObject* was dereferenced");
-  }
-
-  static Error exception(const char *s = nullptr) {
-    return llvm::make_error<PythonException>(s);
-  }
 
   operator PyObject *() const { return m_py_obj; };
 
@@ -275,6 +267,16 @@ public:
 
   StructuredData::ObjectSP CreateStructuredObject() const;
 
+protected:
+  static Error nullDeref() {
+    return llvm::createStringError(llvm::inconvertibleErrorCode(),
+                                   "A NULL PyObject* was dereferenced");
+  }
+  static Error exception(const char *s = nullptr) {
+    return llvm::make_error<PythonException>(s);
+  }
+
+public:
   template <typename... Args>
   Expected<PythonObject> CallMethod(const char *name, const char *format,
                                     Args... args) {
